src/components/PupilViews/PupilProgrammeListing/PupilProgrammeListing.view.tsx

```diff
@@ -162,9 +164,41 @@ export const PupilViewsProgrammeListing = ({
           baseSlug={baseSlug}
           chosenFactors={chosenFactors}
           programmes={programmes}
-          onClick={(f) =>
-            setChosenFactors({ ...chosenFactors, [currentFactor]: f })
-          }
+          onClick={(f) => {
+            setChosenFactors({
+              ...chosenFactors,
+              [currentFactor]: f,
+            });
+          }}
+          onCallback={(f) => {
```

Comment:
this should just be part of onClick but consider refactoring the tracking code into a helper function for readability.